e2e/README.md

```diff
@@ -22,11 +22,53 @@ nodes can be identified by the following container names (kube-master, kube-node
 All tests with the name `SmokeSuite` in the method will execute and the results will be of a junit
 type output to the `e2e/results` directory.
 
+## Run Tests
+Rook Test Framework can be used to set up rook infrastructure and then run tests, or it can be used to run tests against an existing/pre-installed rook ,when users my choose to install kubernetes and rook using their own flavor.  The difference between running test directly and using ```e2e/scripts/smoke_test.sh``` is that rook-test container is not set up. When running tests directly a Kubernetes three node cluster is created using Kubeadm on the Host directly. 
+  
+#### Test paramaters
+The following parameters are available while running tests
+
+ Parameter | Description | Default value
+ --- |--- | --- 
+rook_platform| platform rook needs to be installed on  | kubernetes
+k8s_version  | version of Kubernetes to be installed  | v1.6
+rook_version | rook version/tag to be installed | master-latest 
+install_rook | flag to decide if rook need to be installed | true
+
+if install_rook flag is set to false, then all the other flags are ignored,
+and tests are run without rook being installed and set up. Use this flag to run tests aganist
+a pre-installed/conigured rook. 
+
+
+
+#### Install Rook and run Test
+To install rook and run tests use ```--install_rook=true``` flag.  Then Based on other flags, rook is installed and tests
+are run. 
+e.g.
+```
+go test -run TestObjectStorageSmokeSuite github.com/rook/rook/e2e/tests/smoke --install_rook=true
```

Comment:
can you have the first example be without arguments and describe the default behavior? Then you could have other examples that use the non-default options